python/ray/data/_internal/planner/exchange/aggregate_task_spec.py

```diff
@@ -65,10 +65,15 @@ def map(
         ]
         from ray.data.block import BlockMetadataWithSchema
 
-        meta_with_schema = BlockMetadataWithSchema.from_block(
-            block, stats=stats.build()
-        )
-        return parts + [meta_with_schema]
+        assert len(parts) > 0
+        # Look at the first non-empty block for the schema
+        for b in parts:
+            if b:
+                meta_with_schema = BlockMetadataWithSchema.from_block(
+                    b, stats=stats.build()
+                )
+                return parts + [meta_with_schema]
+        raise RuntimeError("parts should contain at least 1 non-empty block")
```

Comment:
I'm hesitant to add this assertion until we address the issue of having empty blocks floating around let's for now not assume that we can not get empty block